plugin/src/main/java/io/jenkins/plugins/casc/yaml/YamlUtils.java

```diff
@@ -53,58 +47,12 @@ public static Node read(YamlSource source) throws IOException {
         return composer.getSingleNode();
     }
 
-    private static void merge(Node root, Node node, String source) throws ConfiguratorException {
-        if (root.getNodeId() != node.getNodeId()) {
-            // means one of those yaml file doesn't conform to JCasC schema
-            throw new ConfiguratorException(
-                    String.format("Found incompatible configuration elements %s %s", source, node.getStartMark()));
-        }
-
-        switch (root.getNodeId()) {
-            case sequence:
-                SequenceNode seq = (SequenceNode) root;
-                SequenceNode seq2 = (SequenceNode) node;
-                seq.getValue().addAll(seq2.getValue());
-                return;
-            case mapping:
-                MappingNode map = (MappingNode) root;
-                MappingNode map2 = (MappingNode) node;
-                // merge common entries
-                final Iterator<NodeTuple> it = map2.getValue().iterator();
-                while (it.hasNext()) {
-                    NodeTuple t2 = it.next();
-                    for (NodeTuple tuple : map.getValue()) {
-
-                        final Node key = tuple.getKeyNode();
-                        final Node key2 = t2.getKeyNode();
-                        if (key.getNodeId() == NodeId.scalar) {
-                            // We dont support merge for more complex cases (yet)
-                            if (((ScalarNode) key).getValue().equals(((ScalarNode) key2).getValue())) {
-                                merge(tuple.getValueNode(), t2.getValueNode(), source);
-                                it.remove();
-                            }
-                        } else {
-                            throw new ConfiguratorException(
-                                    String.format("Found unmergeable configuration keys %s %s)", source, node.getEndMark()));
-                        }
-                    }
-                }
-                // .. and add others
-                map.getValue().addAll(map2.getValue());
-                return;
-            default:
-                throw new ConfiguratorException(
-                        String.format("Found conflicting configuration at %s %s", source, node.getStartMark()));
-        }
-
-    }
-
     /**
      * Load configuration-as-code model from a set of Yaml sources, merging documents
      */
-    public static Mapping loadFrom(List<YamlSource> sources) throws ConfiguratorException {
+    public static Mapping loadFrom(List<YamlSource> sources, MergeStrategy mergeStrategy) throws ConfiguratorException {
```

Comment:
Seems unnecessary to pass merge strategy here. Properly simpler just to call inside the merge method

Reply:
Actually I think it would be best to move the call to ConfigurationContext constructor.